src/index.js

```diff
@@ -1,6 +1,19 @@
 import { useMemo, useReducer } from "react";
 
-export function createGovernor(initialState = {}, actions = {}) {
+class HookActions {
+  constructor(actions) {
+    for (let key in actions) {
+      this[key] = async (...payload) => {
+        const newState = await actions[key](...payload, this.__state);
+        this.__dispatch({
+          newState: newState
+        });
+      };
```

Comment:
I should be able to do something here like switch from an arrow function to a regular function and bind it to the class so that `this.state` is available in actions, but I haven’t been able to do it yet.